src/main/java/com/mojang/brigadier/builder/ArgumentBuilder.java

```diff
@@ -9,18 +9,26 @@
 import com.mojang.brigadier.tree.CommandNode;
 import com.mojang.brigadier.tree.RootCommandNode;
 
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.List;
 import java.util.function.Predicate;
 
 public abstract class ArgumentBuilder<S, T extends ArgumentBuilder<S, T>> {
     private final RootCommandNode<S> arguments = new RootCommandNode<>();
     private Command<S> command;
-    private Predicate<S> requirement = s -> true;
+    private final List<Predicate<S>> requirements = new ArrayList<>(1);
+    private final List<Predicate<S>> requirementsView;
     private CommandNode<S> target;
     private RedirectModifier<S> modifier = null;
     private boolean forks;
 
+    protected ArgumentBuilder() {
+        requirementsView = Collections.unmodifiableList(requirements);
+        requirements.add(__ -> true);
```

Comment:
This should not be needed anymore, in fact this is why `testCreateBuilder` of ArgumentCommandNode and LiteralCommand node fails (createBuilder will initialize a new builder with this default requirement and add the requirement from the node again, making the builder not equal to the node)

Since you have the same lambda as starting value for the reduce method over there, it should be fine to remove.
https://github.com/Mojang/brigadier/blob/66ab4d6ec9f3eb250ae53d0278ed65a267453aba/src/main/java/com/mojang/brigadier/tree/CommandNode.java#L46